src/Storages/ObjectStorage/DataLakes/Iceberg/IcebergMetadata.cpp

```diff
@@ -707,14 +759,14 @@ std::optional<size_t> IcebergMetadata::totalRows(ContextPtr local_context) const
 
     /// All these "hints" with total rows or bytes are optional both in
     /// metadata files and in manifest files, so we try all of them one by one
-    if (relevant_snapshot->getTotalRows())
+    if (actual_data_snapshot->getTotalRows())
     {
         ProfileEvents::increment(ProfileEvents::IcebergTrivialCountOptimizationApplied);
-        return relevant_snapshot->getTotalRows();
+        return actual_data_snapshot->getTotalRows();
```

Comment:
Why call `actual_data_snapshot->getTotalRows` two times? Better something like this:
```
if (auto total_rows = actual_data_snapshot->getTotalRows(); total_rows != 0)
{
    ProfileEvents::increment(ProfileEvents::IcebergTrivialCountOptimizationApplied);
    return total_rows;
}
```